record/custom_records.go

```diff
@@ -12,8 +14,8 @@ const (
 type CustomSet map[uint64][]byte
 
 // Validate checks that all custom records are in the custom type range.
-func (c CustomSet) Validate() error {
-	for key := range c {
+func (c *CustomSet) Validate() error {
```

Comment:
I don't think this is necessary, since the map type is already a pointer. Also, this code now panics on a `nil` map which I think it didn't before.